test/lint/lint-python-utf8-encoding.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Make sure we explicitly open all text files using UTF-8 (or ASCII) encoding to
+# avoid potential issues on the BSDs where the locale is not always set.
+
+EXIT_CODE=0
+OUTPUT=$(git grep " open(" -- "*.py" | grep -vE "encoding=.(ascii|utf8|utf-8)." | grep -vE "open\([^,]*, ['\"][^'\"]*b[^'\"]*['\"]")
```

Comment:
nit: I would prefer not to use `.` to match `"`

Reply:
It matches also `'`. Imo every other character would be a syntax error in python, so should be fine.

Reply:
@ken2812221 That is intentional to match `'` and `"`.

Reply:
Right, that makes sense.